File: src/za/co/adhd_developers/java/questions/Question16.java
package za.co.adhd_developers.java.questions;

import za.co.adhd_developers.java.tools.Question;

import java.math.BigInteger;

/**
 * Created by Grant on 2017/07/31.
 */
public class Question16 implements Question {

    long answer = 0;

    @Override
    public void doWork() {
        BigInteger digitSum = new BigInteger("2");
        digitSum = digitSum.pow(1000);

        String largeNumber = digitSum.toString();

        for (int i = 0; i < largeNumber.length(); i++) {
            char[] num = new char[1];
            num[0] = largeNumber.charAt(i);

            String strNum = new String(num);

            if (strNum.length() == 1) {
                this.answer = this.answer + Long.parseLong(strNum);
            }
        }
    }

    @Override
    public void printAnswer() {
        System.out.println("=================================");
        System.out.println("Question 16");
        System.out.println("The sum of the numbers from 2^1000 is: " + this.answer);
        System.out.println("=================================");
    }
}

File: src/za/co/adhd_developers/kotlin/questions/Question16.kt
package za.co.adhd_developers.kotlin.questions

import za.co.adhd_developers.kotlin.Question
import java.math.BigInteger

class Question16 : Question {

    var answer = 0

    override fun doWork() {
        var number = BigInteger("2")

        number = number.pow(1000)

        this.answer = number.toString().toCharArray().sumBy { c -> c.toString().toInt() }
    }

    override fun printAnswer() {
        println("=================================")
        println("Question 16")
        println("The sum of the numbers from 2^1000 is: " + this.answer)
        println("=================================")
    }
}